File: Module/PlayWays/src/main/java-room-grab/com/module/playways/grab/room/songmanager/model/GrabWishSongModel.java
package com.module.playways.grab.room.songmanager.model;

import com.common.core.userinfo.model.UserInfoModel;

import java.io.Serializable;

public class GrabWishSongModel extends GrabRoomSongModel implements Serializable {

    private String pID;

    private UserInfoModel suggester;

    public UserInfoModel getSuggester() {
        return suggester;
    }

    public void setSuggester(UserInfoModel suggester) {
        this.suggester = suggester;
    }

    public String getpID() {
        return pID;
    }

    public void setpID(String pID) {
        this.pID = pID;
    }

    @Override
    public String toString() {
        return "GrabRoomSongModel{" +
                "itemName='" + itemName + '\'' +
                ", owner='" + owner + '\'' +
                ", roundSeq=" + roundSeq +
                ", itemID=" + itemID +
                ", playType=" + playType +
                ", challengeAvailable=" + challengeAvailable +
                ", pID=" + pID +
                ", suggester=" + suggester +
                '}';
    }
}

File: Module/PlayWays/src/main/java-songmanage/com/module/playways/songmanager/model/GrabWishSongModel.kt
package com.module.playways.songmanager.model

import com.common.core.userinfo.model.UserInfoModel

import java.io.Serializable

class GrabWishSongModel : GrabRoomSongModel(), Serializable {

    private var pID: String? = null

    var suggester: UserInfoModel? = null

    fun getpID(): String? {
        return pID
    }

    fun setpID(pID: String) {
        this.pID = pID
    }

    override fun toString(): String {
        return "GrabRoomSongModel{" +
                "itemName='" + itemName + '\''.toString() +
                ", owner='" + owner + '\''.toString() +
                ", roundSeq=" + roundSeq +
                ", itemID=" + itemID +
                ", playType=" + playType +
                ", challengeAvailable=" + isChallengeAvailable +
                ", pID=" + pID +
                ", suggester=" + suggester +
                '}'.toString()
    }
}
